src/app/clusters/scenes-server/SceneTableImpl.cpp

```diff
@@ -737,11 +737,17 @@ CHIP_ERROR DefaultSceneTableImpl::SceneSaveEFS(SceneTableEntry & scene)
 {
     if (!HandlerListEmpty())
     {
-        uint8_t clusterCount = 0;
-        clusterId cArray[kMaxClustersPerScene];
-        Span<clusterId> cSpan(cArray);
-        clusterCount = GetClustersFromEndpoint(cArray, kMaxClustersPerScene);
-        cSpan.reduce_size(clusterCount);
+        uint8_t clusterCount = GetClusterCountFromEndpoint();
+        clusterId * cBuffer  = static_cast<clusterId *>(chip::Platform::MemoryCalloc(sizeof(clusterId), clusterCount));
+        VerifyOrReturnError(nullptr != cBuffer, CHIP_ERROR_NO_MEMORY);
+
+        clusterCount = GetClustersFromEndpoint(cBuffer, clusterCount);
+        chip::Platform::MemoryRealloc(cBuffer, clusterCount);
```

Comment:
Why do we need this realloc?  Do we expect this to free up some memory, as opposed to just ending up doing a second allocation alongside the first one, temporarily?  At the very least the reason for this realloc needs to be documented.